### PR TITLE
Remove array elements which are null

### PR DESCRIPTION
**Pull request by heyagosper**

### CODE REVIEW

**Comment by heyagosper** [comment]
this time rebased to master first

File: src/client/app/shared/jsonix/jsonix.service.spec.ts [hunk 1]
```diff
@@ -115,8 +115,56 @@ export function main() {
...
+              "markerArpEastEcc": 8,
+              "markerArpUpEcc": 0,
+              "markerArpNorthEcc": 8,
+              "alignmentFromTrueNorth": 0,
+              "antennaRadomeType": {
+                "value": "SNOW"
+              },
+              "radomeSerialNumber": "",
+              "antennaCableType": "(vendor & type number)",
+              "antennaCableLength": 0,
```
Comment (oehm-smith) [inline src/client/app/shared/jsonix/jsonix.service.spec.ts]: Don't you also need to test for a null in numbers?

File: src/client/app/shared/jsonix/jsonix.service.spec.ts [hunk 2]
```diff
@@ -115,8 +115,56 @@ export function main() {
...
+      }
+    }`;
+
+    fit('should parse valid Json with an array containing a null value', () => {
+      let geodesyMl: string = jsonixService.jsonToGeodesyML(JSON.parse(validJsonWithNullInAnArray));
+      expect(geodesyMl).not.toBeNull();
+      console.log(geodesyMl);
+    });
+
   });
```
Comment (oehm-smith) [inline src/client/app/shared/jsonix/jsonix.service.spec.ts]: Too many blank lines looks funny.

File: src/client/app/shared/jsonix/jsonix.service.ts [hunk 3]
```diff
@@ -69,20 +68,62 @@ export class JsonixService {
...
+   */
+  traverseJsonObject(object : any) : void {
+
+      var type = typeof object;
+
+      if (Array.isArray(object)) {
+          console.log('array');
+          if (object.length === 1 && object[0] === null) {
+              console.log('found an array whose sole element is null - removing the element');
+              object.pop();
```
Comment (oehm-smith) [inline src/client/app/shared/jsonix/jsonix.service.ts]: Is this handled properly in the app?  My experience is that it needs to be [''] (empty string in array).  That might be pre-JSON Pointer which will create the element correctly.  Please confirm.

File: src/client/app/shared/jsonix/jsonix.service.ts [hunk 4]
```diff
@@ -69,20 +68,62 @@ export class JsonixService {
...
+              }, this);
+          }
+      } else {
+          if (type === 'object') {
+              for (var key in object) {
+                  console.log('object key: ', key);
+                  this.traverseJsonObject(object[key]);
+              }
+          } else if (type === 'number') {
+              // check if we have NaN
```
Comment (oehm-smith) [inline src/client/app/shared/jsonix/jsonix.service.ts]: Yet to do?

File: src/client/app/shared/jsonix/jsonix.service.ts [hunk 5]
```diff
@@ -69,20 +68,62 @@ export class JsonixService {
...
+              console.log('number type ', type, ' ', object);
+          } else {
+              // it's a primitive
+              console.log('primitive type ', type, ' ', object);
```
Comment (oehm-smith) [inline src/client/app/shared/jsonix/jsonix.service.ts]: Could you delete these before merging - will cause lots of debug!

File: src/client/app/shared/jsonix/jsonix.service.spec.ts (hunk 1 above)
Comment (heyagosper) [inline src/client/app/shared/jsonix/jsonix.service.spec.ts]: I think it is only in the temperature sensor component in the accuracyDegreesCelcius field. call it a separate ticket

File: src/client/app/shared/jsonix/jsonix.service.spec.ts (hunk 2 above)
Comment (heyagosper) [inline src/client/app/shared/jsonix/jsonix.service.spec.ts]: ok. tidied it up
